fish-rust/Cargo.toml

```diff
@@ -0,0 +1,47 @@
+[package]
+name = "fish-rust"
+version = "0.1.0"
+edition = "2021"
+
+
+[dependencies]
+widestring-suffix = { path = "./widestring-suffix/" }
+
+autocxx = "0.23.1"
+cxx = "1.0"
+errno = "0.2.8"
+inventory = { version = "0.3.3", optional = true}
+lazy_static = "1.4.0"
+libc = "0.2.137"
+nix = "0.25.0"
```

Comment:
This needs to become 

```toml
nix = { version = "0.25.0", default-features = false, features = [ ] }
```

(the features list can be left blank for now, it'll work)

otherwise there are failures linking the executables under BSD:

<details>
<pre>
[5/6] Linking CXX executable fish
FAILED: fish
: && /usr/local/libexec/ccache/clang++ -Wredundant-move -fno-c++-static-destructors -iquote /usr/home/mqudsi/fish-shell/./src -O2 -g -DNDEBUG  CMakeFiles/fish.dir/src/fish.cpp.o -o fish  -Wl,-rpath,/usr/local/lib:  libfishlib.a  libfish_rust.a  /usr/lib/libcurses.so  /usr/local/lib/libtinfo.so  -pthread  /usr/local/lib/libpcre2-32.so  /usr/local/lib/libintl.so && :
ld: error: undefined symbol: mq_open
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_open::hb767fe6feefc2911) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_open::hb767fe6feefc2911) in archive libfish_rust.a
>>> did you mean: kmq_open
>>> defined in: /lib/libc.so.7

ld: error: undefined symbol: mq_unlink
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_unlink::hc12ac9a7732e5462) in archive libfish_rust.a

ld: error: undefined symbol: mq_close
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_close::h42a9979fa4422db4) in archive libfish_rust.a

ld: error: undefined symbol: mq_receive
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_receive::h04907d9bfd2c8a95) in archive libfish_rust.a

ld: error: undefined symbol: mq_send
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_send::hd8c21cd11854cd02) in archive libfish_rust.a

ld: error: undefined symbol: mq_getattr
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_getattr::h4d358e6861ed64a2) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_set_nonblock::h316ea553c44775ab) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_remove_nonblock::hb9b045e36b48ad40) in archive libfish_rust.a

ld: error: undefined symbol: mq_setattr
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_setattr::hec6e064078b9a25f) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_set_nonblock::h316ea553c44775ab) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.1
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.1.rcgu.o:(nix::mqueue::mq_remove_nonblock::hb9b045e36b48ad40) in archive libfish_rust.a

ld: error: undefined symbol: fspacectl
>>> referenced by nix.5cf3894a-cgu.10
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.10.rcgu.o:(nix::fcntl::fspacectl::h28103e804dc77cef) in archive libfish_rust.a
>>> referenced by nix.5cf3894a-cgu.10
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.10.rcgu.o:(nix::fcntl::fspacectl_all::hca1ea434c6d304d2) in archive libfish_rust.a

ld: error: undefined symbol: timer_create
>>> referenced by nix.5cf3894a-cgu.6
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.6.rcgu.o:(nix::sys::timer::Timer::new::h48accb555807a0b6) in archive libfish_rust.a

ld: error: undefined symbol: timer_settime
>>> referenced by nix.5cf3894a-cgu.6
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.6.rcgu.o:(nix::sys::timer::Timer::set::hd9d5a1ad36c6f443) in archive libfish_rust.a

ld: error: undefined symbol: timer_gettime
>>> referenced by nix.5cf3894a-cgu.6
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.6.rcgu.o:(nix::sys::timer::Timer::get::hc1b05f9df3be1863) in archive libfish_rust.a

ld: error: undefined symbol: timer_getoverrun
>>> referenced by nix.5cf3894a-cgu.6
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.6.rcgu.o:(nix::sys::timer::Timer::overruns::he5a9db1fe6333d02) in archive libfish_rust.a

ld: error: undefined symbol: timer_delete
>>> referenced by nix.5cf3894a-cgu.6
>>>               nix-e298e46468831b57.nix.5cf3894a-cgu.6.rcgu.o:(_$LT$nix..sys..timer..Timer$u20$as$u20$core..ops..drop..Drop$GT$::drop::hb9d3f92b12e09718) in archive libfish_rust.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
</pre>
</details>